homeassistant/components/person/__init__.py

```diff
@@ -34,52 +43,181 @@
 })
 
 CONFIG_SCHEMA = vol.Schema({
-    DOMAIN: vol.All(cv.ensure_list, [PERSON_SCHEMA])
+    vol.Optional(DOMAIN): vol.Any(vol.All(cv.ensure_list, [PERSON_SCHEMA]), {})
 }, extra=vol.ALLOW_EXTRA)
 
+_UNDEF = object()
 
-async def async_setup(hass, config):
+
+class PersonManager:
+    """Manage person data."""
+
+    def __init__(self, hass: HomeAssistantType, component: EntityComponent,
+                 config_persons):
+        """Initialize person storage."""
+        self.hass = hass
+        self.component = component
+        self.store = Store(hass, STORAGE_VERSION, STORAGE_KEY)
+        self.storage_data = None
+
+        config_data = self.config_data = OrderedDict()
+        for conf in config_persons:
+            person_id = conf[CONF_ID]
+
+            if person_id in config_data:
+                _LOGGER.error("Found config user with duplicate ID.")
+                continue
+
+            config_data[person_id] = conf
+
+    @property
+    def storage_persons(self):
+        """Iterate over persons stored in storage."""
+        return list(self.storage_data.values())
+
+    @property
+    def config_persons(self):
+        """Iterate over persons stored in config."""
+        return list(self.config_data.values())
+
+    async def async_initialize(self):
+        """Get the person data."""
+        raw_storage = await self.store.async_load()
+
+        if raw_storage is None:
+            raw_storage = {
+                'persons': []
+            }
+
+        storage_data = self.storage_data = OrderedDict()
+
+        for person in raw_storage['persons']:
+            storage_data[person[CONF_ID]] = person
+
+        entities = []
+
+        for person_conf in self.config_data.values():
+            person_id = person_conf[CONF_ID]
+            user_id = person_conf.get(CONF_USER_ID)
+
+            if (user_id is not None
+                    and await self.hass.auth.async_get_user(user_id) is None):
```

Comment:
How to make sure Person component is initializing after `auth` loaded?

Any requirement/dependency of `device_tracker` components?

Reply:
I don't think we need to handle device tracker dependency.

Configuring device tracker for a person is optional, and we only listen for state changes. When those come in we update state.

I think we can assume that state will be synced as long as home assistant is running. Only during startup could there be not sync between person and it's trackers, due to loading one component before the other. But since we use state restore for person and for device trackers, there shouldn't be a difference in state. Am I thinking correctly?

Reply:
Ok, there could be a diff in state if a device tracker new state update comes in between the time that it was loaded and the time that the person component is loaded. Do we need to guard from that?

Reply:
I am wondering, we should probably only restore state if there is no current device tracker state in the state machine? Maybe even drop the restore state completely.

Reply:
What is the motivation?

Reply:
So I updated it now to:
 - restore state when added to hass
 - on HOMEASSISTANT_START event we start tracking device trackers and update to the latest device tracker state

Motivation is that it's up to the device trackers to restore state, and then have person pick that up. Device trackers can better judge if their state is restorable.

Reply:
Ok, sounds good.